### DISCUSSION TITLE
Data model

### DISCUSSION THREAD

**Discussion by wwwronin** [pelicanmapping/rocky#6] [Q&A / Support]
Hi,
  Can you give  a sample description about data model used in rocky. I still do not understand data model and tilekey .
Best regards

**Comment by gwaldron** [comment]
All web mapping applications (Google, Bing, etc.) use a multi-resolution image pyramid to tile up large map datasets. There are tons of resources online that describe the concept (just search for web map tiling). In Rocky, the `TileKey` uniquely identifies a single tile by its Z, X, and Y tile coordinates within the pyramid (where Z = the level of detail).

For more advanced usage, `TileKey` also specifies a `Profile` that describes the tiling scheme itself, including the spatial reference system (SRS), geospatial extents, and how many tiles exist at the lowest level of detail (level zero). You usually don't need to worry about it.

![WMTS](https://github.com/pelicanmapping/rocky/assets/326618/448267f0-bef8-4dbb-ac50-008a01945b1a)

**Reply by wwwronin** [reply]
Thanks a lot.
